### PR TITLE
ENH: add lower gamma

### PR DESCRIPTION
**Pull request by argriffing**
This is mostly copied from the upper gamma implementation. I don't know about a couple of the edge cases that were specially handled by the upper gamma code, so I've left some 'todo' comments. Maybe it's related to branch cuts or something, I'm not sure.

It partially addresses https://github.com/fredrik-johansson/arb/issues/36.

### CODE REVIEW

**Comment by fredrik-johansson** [comment]
This is not using asymptotic expansions, and doesn't support negative integer s. The easiest solution would be to just wrap acb_hypgeom_m.

It's not documented what regularized = 2 does; note that this is not actually the same thing that gamma_upper computes with regularized = 2.

**Comment by argriffing** [comment]
OK I see what you mean, the implementation should look more like http://dlmf.nist.gov/8.5 where `M` is `acb_hypgeom_m`.

> It's not documented what regularized = 2 does; note that this is not actually the same thing that gamma_upper computes with regularized = 2.

If I'm reading the dlmf correctly, I guess that for lower gamma the `s^-1` and `z^s` terms should both be skipped when regularized=2. For the upper gamma only the `z^s` term is skipped.

**Comment by fredrik-johansson** [comment]
Yes, except that regularized=2 in gamma_upper is not really meant to be used by directly; it's just an internal feature that allows expint to be implemented conveniently. It could be renamed to regularized=-1 or regularized=271 to avoid confusion with regularized=2 for gamma_lower, which is a named function in its own right.

File: doc/source/acb_hypgeom.rst [hunk 1]
```diff
@@ -589,6 +589,17 @@ Incomplete gamma functions
     The *singular* version evaluates the finite sum directly and therefore
     assumes that *s* is not too large.
 
+.. function:: void acb_hypgeom_gamma_lower(acb_t res, const acb_t s, const acb_t z, int regularized, slong prec)
+
+    If *regularized* is 0, computes the lower incomplete gamma function
+    `\gamma(s,z)`.
+
+    If *regularized* is 1, computes the regularized lower incomplete
+    gamma function `P(s,z) = \Gamma(s,z) / \Gamma(s)`.
```
Comment (fredrik-johansson) [inline doc/source/acb_hypgeom.rst]: \gamma(s,z)

File: doc/source/acb_hypgeom.rst [hunk 2]
```diff
@@ -589,6 +589,17 @@ Incomplete gamma functions
...
+
+    If *regularized* is 2, computes a further regularized lower incomplete
+    gamma function `\gamma^{*}(s,z) = z^{-s} P(s,z)`.
```
Comment (fredrik-johansson) [inline doc/source/acb_hypgeom.rst]: Missing the / s. Could also write that this is equal to M(...).

**Comment by argriffing** [comment]
I've updated this PR to just wrap acb_hypgeom_m, but I'm not sure if the details related to singularities, holomorphic extensions, or branch cuts are correct. Some DLMF identities are tested.

File: doc/source/acb_hypgeom.rst [hunk 3]
```diff
@@ -589,6 +589,17 @@ Incomplete gamma functions
     The *singular* version evaluates the finite sum directly and therefore
     assumes that *s* is not too large.
 
+.. function:: void acb_hypgeom_gamma_lower(acb_t res, const acb_t s, const acb_t z, int regularized, slong prec)
+
+    If *regularized* is 0, computes the lower incomplete gamma function
+    `\gamma(s,z)`.
```
Comment (fredrik-johansson) [inline doc/source/acb_hypgeom.rst]: Could give the definition in terms of 1F1 here.

File: doc/source/acb_hypgeom.rst (hunk 1 above)
Comment (argriffing) [inline doc/source/acb_hypgeom.rst]: fixed

File: doc/source/acb_hypgeom.rst (hunk 3 above)
Comment (argriffing) [inline doc/source/acb_hypgeom.rst]: I've added this, copying some markup from the upper gamma docs.

**Pull request merged**